Lots of NCRC HowTo TLC

milljm opened the pull request:
More than just being incorrect, there were plenty of areas that needed some TLC...

Closes #22444

Commit:
Lots of NCRC HowTo TLC

More than just being incorrect, there were plenty of areas that needed
some TLC...

Closes #22444

cticenhour commented on modules/doc/content/ncrc/ncrc_local_binary.md:
````diff
@@ -42,144 +42,176 @@ optional arguments:
 
-  ```bash
-  $> ncrc list
+# Use 'ncrc search name-of-application' to list more detail
+# NCRC applications available:
````
Did you mean for this to be the highest level of heading?

milljm replied:
That happens to be a comment being generated by the `ncrc` script:

```pre
\```bash
 $> ncrc list

# Use 'ncrc search name-of-application' to list more detail
# NCRC applications available:

pronghorn
bison
griffin
sabertooth
relap7
\```
```

milljm replied:
I've been re-doing the indentation going on. I don't much care for the itemized list justification going on, for only half the page...

cticenhour replied:
Oh sorry, I completely missed that it was part of the output..... You can disregard this one. I liked how you moved away from so many itemized lists as well.

cticenhour commented on modules/doc/content/ncrc/ncrc_local_binary.md:
````diff
@@ -42,144 +42,176 @@ optional arguments:
-  ncrc-griffin              2021_09_25         build_0  ncrc-applications
-  ncrc-griffin              2021_10_02         build_0  ncrc-applications
-  ```
+Installing an NCRC application requires authentication. If you have sufficient privileges the package installation will commence:
````
```suggestion
Installing an NCRC application requires authentication. If you have sufficient privileges, the package installation will commence:
```

cticenhour commented on modules/doc/content/ncrc/ncrc_local_binary.md:
```diff
@@ -42,144 +42,176 @@ optional arguments:
+##### Install a specific NCRC application version
 
-- Install a specific version
+Don't want to use the latest? For whatever the reason, you can install a specific version of an available NCRC application in the following way:
```
```suggestion
Don't want to use the latest version? For whatever the reason, you can install a specific version of an available NCRC application in the following way:
```

cticenhour commented on modules/doc/content/ncrc/ncrc_local_binary.md:
```diff
@@ -42,144 +42,176 @@ optional arguments:
-The NCRC script being a wrapper tool, is unable to perform such a function. The user must deactivate the environment and remove that environment using the appropiate conda commands instead:
+##### Uninstalling NCRC application
+
+The NCRC script being a wrapper tool, is unable to perform such a function. The user must deactivate the environment
```
```suggestion
The NCRC script, being a wrapper tool, is unable to perform such a function. The user must deactivate the environment
```

GiudGiud commented on modules/doc/content/ncrc/ncrc_local_binary.md:
```diff
@@ -42,144 +42,176 @@ optional arguments:
+##### View Documentation
+
+Documentation exists as part of the binary installation process. You can access documentation by instructing your favorite web browser to open the index.html file.
+We have made finding the location of this file via an environment variable, prefix with the NCRC application name:
```
```suggestion
We have made finding the location of this file easier via an environment variable, prefixed with the NCRC application name:
```

milljm replied:
Its very much necessary in fact. Griffin for example my need/require a different set of libraries/versions of libMesh at the time the application was built.

GiudGiud replied:
ok let's say it explicitly somewhere then? 

unless I missed it

milljm replied:
Well, actually I am wrong in how I originally defended the requirement. But they still should be separated. The ncrc assumes the application name is also the environment name.

milljm replied:
Yeah I'll make mention of this in the first alert on the page. In my defense, the ncrc script itself will actually warn/error when the user tries to use it while not in the base environment (except during upgrades).

cticenhour replied:
More content regarding separate environments for each app was added in subsequent commits. I think this is resolved.

cticenhour commented on modules/doc/content/ncrc/ncrc_local_binary.md:
````diff
@@ -42,144 +42,176 @@ optional arguments:
+```
 
-The NCRC script being a wrapper tool, is unable to perform such a function. The user must deactivate the environment and remove that environment using the appropiate conda commands instead:
+##### Uninstalling NCRC application
````
```suggestion
##### Uninstalling an NCRC application
```

cticenhour commented on modules/doc/content/ncrc/ncrc_local_binary.md:
````diff
@@ -42,144 +42,176 @@ optional arguments:
-  $> conda activate bison
-  $ (bison) > open $bison_DOCS
-  ```
+Where applicable, you only need to activate the environment once. Each example below assume the environment has not yet been activated.
````
```suggestion
Where applicable, you only need to activate the environment once. Each example below assumes the environment has not yet been activated.
```

cticenhour commented on modules/doc/content/ncrc/ncrc_local_binary.md:
```diff
@@ -42,144 +42,176 @@ optional arguments:
+
+Some applications have different inputs that can be copied and executed. Each application can use whatever naming convention
+they wish, so listing all possibilities is impossible. However, any runnable input directory will have a file called `testroot`
+residing in it. The following *crazy* command will return those possibilities (using Bison as an example):
```
🤪

milljm replied:
yeah... hahahaha any help here in making this... better described is appreciated!

cticenhour replied:
Added another comment with a wording suggestion that you can take or leave. I appreciate that you are giving a resource for the user to find what those input categories actually are.

milljm replied:
Yeah, I then need to add what to actually do with these. Basically another example of how one runs tests.

GiudGiud reviewed (commented):
do you advise to use a different environment for every app?

cticenhour commented on modules/doc/content/ncrc/ncrc_local_binary.md:
```diff
@@ -42,144 +42,176 @@ optional arguments:
+
+##### View Documentation
+
+Documentation exists as part of the binary installation process. You can access documentation by instructing your favorite web browser to open the index.html file.
```
```suggestion
Documentation exists as part of the binary installation process. You can access documentation by instructing your favorite web browser to open the `index.html` file.
```

cticenhour commented on modules/doc/content/ncrc/ncrc_local_binary.md:
```diff
@@ -42,144 +42,176 @@ optional arguments:
+##### View Documentation
+
+Documentation exists as part of the binary installation process. You can access documentation by instructing your favorite web browser to open the index.html file.
+We have made finding the location of this file via an environment variable, prefix with the NCRC application name:
```
```suggestion
We have made finding the location of this file via an environment variable, prefixed with the NCRC application name:
```

cticenhour commented on modules/doc/content/ncrc/ncrc_local_binary.md:
```diff
@@ -42,144 +42,176 @@ optional arguments:
+
+##### Some Applications contain more than tests
+
+Some applications have different inputs that can be copied and executed. Each application can use whatever naming convention
+they wish, so listing all possibilities is impossible. However, any runnable input directory will have a file called `testroot`
+residing in it. The following *crazy* command will return those possibilities (using Bison as an example):
```
```suggestion
Some applications have different inputs that can be copied and executed using the `copy-inputs` flag. Each application can use whatever naming convention
they wish for these input categories, so listing all possibilities is impossible. However, any runnable input directory will have a file called `testroot`
residing in it, in which the test categories are listed. The following piped command sequence will return those possibilities (using Bison as an example):
```

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1208966/) on 4284be4 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/22446/site)

This comment will be updated on new commits.

Commit:
Fix typos, address comments, template apps

Darkmode the menu!

Template the applications for a more personalized experience.

Rename the Documentation, INL HPC to INL Applications.
INL HPC made little sense. And we seem to have HPC/INL, HPC stuff
scattered about. This PR is by no means finished.

Address comments

milljm commented on framework/doc/content/css/moose.css:
```diff
@@ -610,7 +611,7 @@ footer.page-footer {
 
 .container{
-    width:70%;
+    width:80%;
```
I think an additional 10% usable space looks great...

Commit:
Missed a few things

Commit:
more small changes
